src/Formatter/CrossReferencesConfigurator.php

```diff
@@ -100,11 +102,9 @@ protected function configureCrossReferenceShort(Configurator $config): void
         $tag->template = '
             <xsl:choose>
                 <xsl:when test="@unknown = 1">
-                    <span class="DiscussionLink DiscussionUnknown">
+                    <a href="{$DISCUSSION_URL}{@id}" class="DiscussionLink">
```

Comment:
I don't want to add the link if the discussion is unknown. This will lead to bad experiences from the user getting 404 errors (Flarum does not differentiate between unauthorised and not found). And it is overall not good to have internal links leading to 404 errors.

Reply:
But you don't know if it is 404 - permissions detection may be incorrect and link could actually work. In this way extension improves behavior for safe cases and do not break anything in uncertain scenarios.

Reply:
Isn't the Guest fallback enough for most use cases? And isn't [this change](https://github.com/club-1/flarum-ext-cross-references/pull/31/files#diff-18af1e3972585e1b07ba509ff590f4281ac1232144731d14c87a4cd5a0f74e85R82) enough to fix the detection in your case?
I don't mind marking this extension incompatible with others for some time while the render/parse calls are incomplete. For instance `flarum/mentions` just crashes in these cases.